lib/taskana-core/src/main/java/org/taskana/impl/persistence/ClassificationQueryImpl.java

```diff
@@ -0,0 +1,159 @@
+
+import org.apache.ibatis.session.RowBounds;
+import org.taskana.TaskanaEngine;
+import org.taskana.impl.TaskanaEngineImpl;
+import org.taskana.model.Classification;
+import org.taskana.persistence.ClassificationQuery;
+
+/**
+ * Implementation of ClassificationQuery interface.
+ * @author EH
+ */
+public class ClassificationQueryImpl implements ClassificationQuery {
+
+    private static final String LINK_TO_MAPPER = "org.taskana.model.mappings.QueryMapper.queryClassification";
+    private TaskanaEngineImpl taskanaEngine;
+    private String tenantId;
+    private String[] parentClassificationId;
+    private String[] category;
+    private String[] type;
+    private String[] name;
+    private String description;
+    private int[] priority;
+    private String[] serviceLevel;
+
+    public ClassificationQueryImpl(TaskanaEngine taskanaEngine) {
+        this.taskanaEngine = (TaskanaEngineImpl) taskanaEngine;
+    }
+
+    @Override
+    public ClassificationQuery tenantId(String tenantId) {
+        this.tenantId = tenantId;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery parentClassification(String... parentClassificationId) {
+        this.parentClassificationId = parentClassificationId;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery category(String... category) {
+        this.category = category;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery type(String... type) {
+        this.type = type;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery name(String... name) {
+        this.name = name;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery descriptionLike(String description) {
+        this.description = description;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery priority(int... priorities) {
+        this.priority = priorities;
+        return this;
+    }
+
+    @Override
+    public ClassificationQuery serviceLevel(String... serviceLevel) {
+        this.serviceLevel = serviceLevel;
+        return this;
+    }
+
+    @Override
+    public List<Classification> list() {
+        return taskanaEngine.getSession().selectList(LINK_TO_MAPPER, this);
+    }
+
+    @Override
+    public List<Classification> list(int offset, int limit) {
+        RowBounds rowBounds = new RowBounds(offset, limit);
+        return taskanaEngine.getSession().selectList(LINK_TO_MAPPER, this, rowBounds);
+    }
+
+    @Override
+    public Classification single() {
+        return taskanaEngine.getSession().selectOne(LINK_TO_MAPPER, this);
+    }
+
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    public void setTenantId(String tenantId) {
+        this.tenantId = tenantId;
+    }
+
+    public String[] getParentClassificationId() {
+        return parentClassificationId;
+    }
+
+    public void setParentClassificationId(String[] parentClassificationId) {
```

Comment:
Ich benötige diese Getter und Setter für Mybartis, deswegen sind die auch in der Impl und nicht im Interface ...